assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js

```diff
@@ -61,10 +61,16 @@ export default function PublicationApprovedOverlayNotification() {
 			( select ) =>
 				select( MODULES_READER_REVENUE_MANAGER ).getSettings() || {}
 		);
-	const initialPublicationOnboardingStateChanged = useRef(
-		publicationOnboardingStateChanged
+
+	// Check if settings has been resolved.
+	const hasSettingsResolved = useSelect( ( select ) =>
+		select( MODULES_READER_REVENUE_MANAGER ).hasFinishedResolution(
+			'getSettings'
+		)
```

Comment:
Extremely nitpicky, but I believe `hasResolvedSettings` sounds more appropriate in this context, as otherwise, we'd want to change it to `haveSettingsResolved` as "settings" is "plural". `hasResolvedSettings` is also resonant to other similar variables in the codebase.

Additionally, I don't think the comment is necessary here as it is quite self-explanatory.

```suggestion
	const hasResolvedSettings = useSelect( ( select ) =>
		select( MODULES_READER_REVENUE_MANAGER ).hasFinishedResolution(
			'getSettings'
		)
```